bin/calculate_perc_mapping.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import argparse
+from collections import defaultdict
+import csv
+import json
+from pathlib import Path
+from typing import Optional, Set, Tuple, Dict
+
+VERSION = "1.0.0"
+
+
+def main(
+    hbs_path: Path,
+    gene_counts_path: Path,
+    strandedness: Optional[str],
+    out_json: Optional[Path],
+    verbose: bool,
+):
+
+    target_genes = get_target_genes(hbs_path)
+
+    if strandedness not in ["forward", "reverse"] and strandedness is not None:
+        raise ValueError(
+            f"strandedness option should be forward, reverse or not assinged. Found: {strandedness}"
+        )
+
+    (nbr_genes, total_count, target_genes_counts) = summarize_gene_counts(
+        gene_counts_path, target_genes, strandedness, verbose
+    )
+
+    metrics = dict()
+    metrics["nbr_genes"] = nbr_genes
+    metrics["total_count"] = total_count
+    metrics["average_count"] = total_count / nbr_genes
+    metrics["target_genes_count"] = sum(list(target_genes_counts.values()))
+    metrics["target_genes_frac"] = metrics["target_genes_count"] / metrics["total_count"]
+    metrics["per_target_gene_count"] = target_genes_counts
+
+    if verbose or out_json is None:
+        print(json.dumps(metrics))
+
+    if out_json is not None:
+        with out_json.open("w") as out_fh:
+            out_fh.write(json.dumps(metrics))
```

Comment:
```suggestion
    if out_json:
        with out_json.open("w") as out_fh:
            json.dump(metrics, out_fh)
```
json.dump would write the JSON directly to a file object, avoiding the intermediate string, it's slightly faster for larger datasets.